### PR TITLE
hpc: Add munge sanity test and refactor multiple used methods to hpc-baseclass

### PR DESCRIPTION
**Pull request by Soulofdestiny**

### CODE REVIEW

**Comment by Soulofdestiny** [comment]
Verification runs:
master: http://drnek.arch.suse.de/tests/388
slave: http://drnek.arch.suse.de/tests/389

File: tests/hpc/munge_master.pm [hunk 1]
```diff
@@ -11,38 +11,46 @@
...
-    type_string $cmd;
-    send_key "ret";
-    assert_screen 'password-prompt';
-    type_password;
-    send_key "ret";
-}
-
 sub run() {
+    my $self     = shift;
+    my $host_ip  = "10.0.2.10/15";
```
Comment (asmorodskyi) [inline tests/hpc/munge_master.pm]: please extract it to variable

File: tests/hpc/munge_slave.pm [hunk 2]
```diff
@@ -11,24 +11,41 @@
...
     assert_script_run('systemctl enable munge.service');
     assert_script_run('systemctl start munge.service');
-    barrier_wait('MUNGE_SERVICE_START');
-    barrier_wait('TEST_END');
+    barrier_wait("service_enabled");
+
+    mutex_lock('munge_done');
+
+    # wait for master to finish
+    wait_for_children;
```
Comment (asmorodskyi) [inline tests/hpc/munge_slave.pm]: check that you can remove this line

**Comment by asmorodskyi** [comment]
All barriers/mutexes suppose to be in uppercase

**Comment by Soulofdestiny** [comment]
updated, waiting for new verification runs

**Pull request merged**